_posts/2020-03-24-dask-developer-workshop.markdown

```diff
@@ -0,0 +1,90 @@
+When investigating the Dask.optimization module, we saw that code to optimize memory usage was already there. In practice, however, the graphs are not executed in such an order because of other constraints during execution.
+
+A Dask user at the workshop facing this issue told us that they worked around this by injecting dummy dependencies into Dask graphs. These dependencies acted as "choke-holds" for certain types of tasks, in order to improve the memory usage during execution.
+
+This lack of optimization, also referred to as memory back-pressure, does not only impact memory usage but also increases resource consumption. We would greatly benefit from an implementation which addresses this issue.
+
+## Looking back
+
+### Results from working sessions
+Lucas had the pleasure to collaborate with John Lee on writing down a benchmark to enable [work stealing for tasks with restrictions](https://github.com/Dask/distributed/pull/3069). He also got valuable input from Tom Augspurger on dask internals while debugging a Dask.dataframe bug involving categoricals.
+
+However, arguably the most useful work we did was the implementation of a semaphore in Dask.distributed.
+We need to be able to rate-limit the access of the computation clusters to certain resources such as production databases. Migrating workflows which were still heavily dependent on accessing the database to the distributed scheduler was not possible without a semaphore implementation.
+By working on the semaphore during the workshop, we unblocked a Blue Yonder product team in its migration to Dask.
+Coincidentally, we also talked with other attendees of the workshop whom were interested in such a functionality.
+This is something we did not forget; despite the jetlag, we left D.C. with a good chunk of the implementation necessary for a semaphore. This finally got [released](https://github.com/Dask/distributed/commit/2129b740c1e3f524e5ba40a0b6a77b239d4c1f94) with distributed 2.14.0.
+
+### Interaction with the community
+Being able to share our issues with the Dask community and discuss potential ways of improvement with expert users and core developers was extremely valuable. Additionally, this interaction gave us a wider perspective on the current status of Dask, the ecosystem around it, and what we can expect in the future.
+
+We were also reminded once again by the value of open-source software. 
+There was a clear synergy in terms of intended functionality between [Kartothek](https://github.com/JDASoftwareGroup/kartothek) and other tools in the ecosystem such as Arrow/Parquet and Dask.dataframe I/O and partitioning.
+We are sharing responsibility of Dask with the community; by committing to this project we are accessing a broad pool of developers.
```

Comment:
This sentence and the one below - I do not understand the intention. What exactly do you want to express?

Reply:
Mostly pro-OSS propaganda/appreciation

Reply:
Would you phrase this differently?

Reply:
> Mostly pro-OSS propaganda/appreciation
Lol
> Would you phrase this differently?
Erm.. I'd suggest something like:

```suggestion
Dask is owned and driven [hope this is true] by a wide community of skilled and passionate developers. For us, therefore, committing to Dask implies not only that we get access to a great piece of software but also that we get to collaboratively helps shape the future of the project. 
```

Note, that that's a first draft suggestion. Will post further suggestions here if I come up with better ones.

Reply:
It's owned and driven primarily by Matt Rocklin AFAIK, so that is not entirely accurate

Reply:
Lol, okay. Will think of how to re-rephrase.

Reply:
I am sure, Matthew likes to read, that there is a strong community instead of a one man show...no matter what the truth is...we should do him the favor and say it is a strong community, so I like Kishs suggestion

Reply:
```suggestion
Dask is driven by a wide community of skilled and passionate developers. For us, therefore, committing to Dask implies not only that we get access to a great piece of software but also that we get to collaboratively help shape future developments. 
```